### PR TITLE
Move fieldFilter (free function) to Filter.create()

### PR DESCRIPTION
**Pull request by rsgowman**
This is a refactoring to unify filter creation across platforms.

### CODE REVIEW

File: packages/firestore/src/core/query.ts [hunk 1]
```diff
@@ -434,12 +463,12 @@ export class RelationOp {
   }
 }
 
-export class RelationFilter implements Filter {
+export class RelationFilter extends Filter {
   constructor(
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/query.ts]: You can omit constructors that match the super class constructor (here and everywhere)..

File: packages/firestore/src/core/query.ts [hunk 2]
```diff
@@ -390,10 +390,39 @@ export class Query {
...
+  static create(
+    field: FieldPath,
+    op: RelationOp,
+    value: FieldValue
+  ): Filter {
+    if (value.isEqual(NullValue.INSTANCE)) {
+      if (op !== RelationOp.EQUAL) {
+        throw new FirestoreError(
+          Code.INVALID_ARGUMENT,
+          'Invalid query. You can only perform equals ' + 'comparisons on null.'
```
Comment (schmidt-sebastian) [inline packages/firestore/src/core/query.ts]: Do you mind removing the extraneous string concatenation?

Comment (rsgowman) [reply packages/firestore/src/core/query.ts]: Done.

File: packages/firestore/src/core/query.ts (hunk 1 above)
Comment (rsgowman) [reply packages/firestore/src/core/query.ts]: Don't you need to keep them to ensure (in this case) field, op, and value are present? Or in other words, these ctors don't match the parent ctor.

Comment (schmidt-sebastian) [reply packages/firestore/src/core/query.ts]: Oh, yes, that's true. The constructors are much more powerful at second glance.

**Pull request merged**